File: flutter/lib/theme/app_theme.dart
import 'package:flutter/material.dart';
import 'dart:ui';

/// Apple-style theme configuration with glassmorphism support
class AppTheme {
  // Primary Colors
  static const Color primaryBlue = Color(0xFF007AFF);
  static const Color primaryBlueDark = Color(0xFF0051D5);

  // Accent Colors
  static const Color accentGreen = Color(0xFF34C759);
  static const Color accentOrange = Color(0xFFFF9500);
  static const Color accentRed = Color(0xFFFF3B30);

  // Glass Colors
  static Color glassLight = Colors.white.withOpacity(0.15);
  static Color glassDark = Colors.black.withOpacity(0.3);
  static Color glassBorder = Colors.white.withOpacity(0.3);

  // Gradients
  static const LinearGradient primaryGradient = LinearGradient(
    begin: Alignment.topLeft,
    end: Alignment.bottomRight,
    colors: [primaryBlue, primaryBlueDark],
  );

  static LinearGradient glassGradient = LinearGradient(
    begin: Alignment.topLeft,
    end: Alignment.bottomRight,
    colors: [
      Colors.white.withOpacity(0.2),
      Colors.white.withOpacity(0.1),
    ],
  );

  // Multi-layer Shadow Pattern
  static List<BoxShadow> multiLayerShadow = [
    BoxShadow(
      color: Colors.white.withOpacity(0.5),
      offset: const Offset(-4, -4),
      blurRadius: 15,
      spreadRadius: 1,
    ),
    BoxShadow(
      color: Colors.black.withOpacity(0.2),
      offset: const Offset(4, 4),
      blurRadius: 15,
      spreadRadius: 1,
    ),
  ];

  // Card Shadows
  static List<BoxShadow> cardShadow = [
    BoxShadow(
      color: Colors.black.withOpacity(0.1),
      offset: const Offset(0, 2),
      blurRadius: 4,
    ),
    BoxShadow(
      color: Colors.black.withOpacity(0.08),
      offset: const Offset(0, 8),
      blurRadius: 16,
    ),
  ];

  // Button Shadow
  static List<BoxShadow> buttonShadow = [
    BoxShadow(
      color: primaryBlue.withOpacity(0.3),
      offset: const Offset(0, 8),
      blurRadius: 20,
    ),
  ];

  // Elevation Shadow
  static List<BoxShadow> elevationShadow = [
    BoxShadow(
      color: Colors.black.withOpacity(0.1),
      offset: const Offset(0, 10),
      blurRadius: 20,
    ),
  ];

  // Light Theme
  static ThemeData lightTheme = ThemeData(
    brightness: Brightness.light,
    primaryColor: primaryBlue,
    scaffoldBackgroundColor: Colors.grey[100],
    colorScheme: const ColorScheme.light(
      primary: primaryBlue,
      secondary: accentGreen,
      error: accentRed,
    ),
    fontFamily: 'Inter',
    textTheme: const TextTheme(
      displayLarge: TextStyle(fontSize: 32, fontWeight: FontWeight.w700, color: Colors.black),
      displayMedium: TextStyle(fontSize: 24, fontWeight: FontWeight.w600, color: Colors.black),
      bodyLarge: TextStyle(fontSize: 16, fontWeight: FontWeight.w400, color: Colors.black87),
      bodyMedium: TextStyle(fontSize: 14, fontWeight: FontWeight.w400, color: Colors.black87),
    ),
  );

  // Dark Theme
  static ThemeData darkTheme = ThemeData(
    brightness: Brightness.dark,
    primaryColor: primaryBlue,
    scaffoldBackgroundColor: Colors.black,
    colorScheme: const ColorScheme.dark(
      primary: primaryBlue,
      secondary: accentGreen,
      error: accentRed,
    ),
    fontFamily: 'Inter',
    textTheme: const TextTheme(
      displayLarge: TextStyle(fontSize: 32, fontWeight: FontWeight.w700, color: Colors.white),
      displayMedium: TextStyle(fontSize: 24, fontWeight: FontWeight.w600, color: Colors.white),
      bodyLarge: TextStyle(fontSize: 16, fontWeight: FontWeight.w400, color: Colors.white70),
      bodyMedium: TextStyle(fontSize: 14, fontWeight: FontWeight.w400, color: Colors.white70),
    ),
  );

  // Animation Curves
  static const Curve easeInOutCubic = Curves.easeInOutCubic;
  static const Curve elasticOut = Curves.elasticOut;
  static const Curve fastOutSlowIn = Curves.fastOutSlowIn;

  // Border Radius
  static BorderRadius cardRadius = BorderRadius.circular(20);
  static BorderRadius buttonRadius = BorderRadius.circular(14);
  static BorderRadius modalRadius = const BorderRadius.vertical(top: Radius.circular(24));
  static BorderRadius pillRadius = BorderRadius.circular(25);
  static BorderRadius navBarRadius = BorderRadius.circular(30);
}

File: react-native/src/theme/AppTheme.js
/**
 * Apple-style theme configuration with glassmorphism support
 */

export const AppTheme = {
  // Primary Colors
  colors: {
    primaryBlue: '#007AFF',
    primaryBlueDark: '#0051D5',

    // Background Colors
    darkBackground: '#000000',
    lightBackground: '#F5F5F5',

    // Secondary Colors
    secondaryPurple: '#7B68EE',
    errorRed: '#FF3B30',

    // Accent Colors
    accentGreen: '#34C759',
    accentOrange: '#FF9500',
    accentRed: '#FF3B30',

    // Glass Colors
    glassLight: 'rgba(255, 255, 255, 0.15)',
    glassDark: 'rgba(0, 0, 0, 0.3)',
    glassBorder: 'rgba(255, 255, 255, 0.3)',
  },

  // Gradients
  gradients: {
    primary: ['#007AFF', '#0051D5'],
    glass: ['rgba(255, 255, 255, 0.2)', 'rgba(255, 255, 255, 0.1)'],
    instagram: ['#f09433', '#e6683c', '#dc2743', '#cc2366', '#bc1888'],
  },

  // Shadows
  shadows: {
    multiLayer: {
      shadowColor: '#000',
      shadowOffset: { width: 0, height: 4 },
      shadowOpacity: 0.15,
      shadowRadius: 15,
      elevation: 10,
    },
    card: {
      shadowColor: '#000',
      shadowOffset: { width: 0, height: 2 },
      shadowOpacity: 0.1,
      shadowRadius: 8,
      elevation: 5,
    },
    button: {
      shadowColor: '#007AFF',
      shadowOffset: { width: 0, height: 8 },
      shadowOpacity: 0.3,
      shadowRadius: 20,
      elevation: 8,
    },
    elevation: {
      shadowColor: '#000',
      shadowOffset: { width: 0, height: 10 },
      shadowOpacity: 0.15,
      shadowRadius: 20,
      elevation: 12,
    },
  },

  // Border Radius
  borderRadius: {
    card: 20,
    button: 14,
    modal: 24,
    pill: 25,
    navBar: 30,
  },

  // Spacing
  spacing: {
    xs: 4,
    sm: 8,
    md: 12,
    lg: 16,
    xl: 20,
    xxl: 24,
  },

  // Typography
  typography: {
    displayLarge: {
      fontSize: 32,
      fontWeight: '700',
      lineHeight: 40,
    },
    displayMedium: {
      fontSize: 24,
      fontWeight: '600',
      lineHeight: 32,
    },
    bodyLarge: {
      fontSize: 16,
      fontWeight: '400',
      lineHeight: 24,
    },
    bodyMedium: {
      fontSize: 14,
      fontWeight: '400',
      lineHeight: 20,
    },
    caption: {
      fontSize: 12,
      fontWeight: '400',
      lineHeight: 16,
    },
  },

  // Animation Timing
  animation: {
    fast: 150,
    normal: 300,
    slow: 600,
    flip: 800,
  },
};

export default AppTheme;
